application/src/main/java/org/thingsboard/server/service/install/SqlDatabaseUpgradeService.java

```
@@ -116,7 +167,108 @@ public void upgradeDatabase(String fromVersion) throws Exception {
+                                                    Map<String, PersistedAlarmState> alarmStates = new HashMap<>();
+                                                    DeviceId deviceId = new DeviceId(UUID.fromString(stateNode.get("entity_id").asText()));
+                                                    RuleNodeId ruleNodeId = new RuleNodeId(UUID.fromString(stateNode.get("rule_node_id").asText()));
+                                                    boolean debugMode = stateNode.get("debug_mode").asBoolean();
+
+                                                    PersistedEntityState persistedEntityState = JacksonUtil.fromString(stateNode.get("state_data").asText(), PersistedEntityState.class);
+                                                    persistedEntityState.setTenantId(tenantId);
+                                                    persistedEntityState.setEntityId(deviceId);
+
+                                                    persistedEntityState.getAlarmStates().forEach((id, alarmState) -> {
+                                                        String newId = alarmRuleIdMapping.get(id);
+                                                        alarmState.setLastRuleNodeId(ruleNodeId);
+                                                        alarmState.setLastRuleChainId(ruleChainId);
+                                                        alarmState.setLastRuleNodeDebugMode(debugMode);
+                                                        alarmStates.put(newId, alarmState);
+                                                    });
+
+                                                    persistedEntityState.setAlarmStates(alarmStates);
+
+                                                    stateStore.get().put(persistedEntityState);
+                                                });
+                                            }
+                                        } catch (Exception e) {
+                                        }
```

Why ignore the error?